include/crm/common/results.h

```diff
@@ -46,6 +46,16 @@ extern "C" {
         }                                                                    \
     } while(0)
 
+/*
+ * Error domains for use with g_set_error.
+ */
+
+GQuark pcmk_cib_error_quark(void);
+GQuark pcmk_working_set_error_quark(void);
+
+#define PCMK_CIB_ERROR              pcmk_cib_error_quark()
+#define PCMK_WORKING_SET_ERROR      pcmk_working_set_error_quark()
+
```

Comment:
I guess the docs are a little unclear here.  I took "domain indicates the module the error-reporting function is located in" and "The error domain is called <NAMESPACE>_<MODULE>_ERROR, for example G_SPAWN_ERROR or G_THREAD_ERROR" to mean that PCMK is the namespace, and the module is some sub-portion of pacemaker.  I don't really have any strong opinions about this, beyond the fact that what I was doing with G_OPTION_ERROR everywhere is incorrect.

I guess I don't see why we can't also use G_OPTION_ERROR, unless it really is just for glib's internal use.  I didn't see any indication that's the case in the docs.  It just says it's the error domain for option parsing.  But it's not a big deal to also convert these to something else.

Finally, I don't even know where these error domains get used.  Are they ever publically visible?  I haven't found any explanation as to why we really care about error domain.  Maybe they appear in a log file somewhere.

Reply:
Given the example code from the docs:

    g_set_error (error,
                         FOO_ERROR,                 // error domain
                         FOO_ERROR_BLAH,      // error code
                         "Failed to open file: %s",  // error message format string
                         g_strerror (saved_errno));

It seems to me that the error code is going to be some integer, and the recipient of the error does not necessarily know how to interpret that integer. Thus error domain is a way to specify that, so you want one error domain per set of possible error codes. The main use of the domain is g_error_matches().

The documentation section describing the error domain as <NAMESPACE>_<MODULE>_ERROR is specifically about glib errors (e.g. also giving GThreadError as an example of error code naming). So I think one error domain for all standard Pacemaker return codes makes sense. Since we also have exit codes, I could see having two domains, e.g. PCMK_RC_ERROR and PCMK_EC_ERROR (or PCMK_EXIT_ERROR), for future-proofing.

Reply:
Okay, this makes sense to me.

Reply:
Is it fine if these go directly into include/crm/common/internal.h, or should I start a new header file?

Reply:
internal.h is fine